### ISSUE TITLE
VRP solution with metaheuristics taking too much time

### ISSUE THREAD

**Issue by anis-coding** [google/or-tools#1297]
I have turned a routing problem into a vrp problem and then developed a solution using the code provided by the VRP guide on ortools.

When I run it on 12 nodes, it's fast but on 72 nodes it takes too much time and my laptop start overheating

I have tried to give a time limit  of 2 hours with no success and when I check the status of solver it prints 3 which correspend to `ROUTING_FAIL_TIMEOUT: Time limit reached before finding a solution.`
For my machine benchmarks, I'm using Ubuntu on an intel I5 computer dual booted with Windows, and a clock speed of around 1.8 GHz.
Even though I know that the complexity of vrp problems grows exponentially, I believe for 70 - 80 nodes, it would still be reasonable to solve.

Here's the solve part of my code, mostly pasted from the vrp code on the guide:
```
def main():
    """Solve the CVRP problem."""
    # Instantiate the data problem.
    data = create_data_model()

    # Create the routing index manager.
    manager = pywrapcp.RoutingIndexManager(
        len(data['distance_matrix']), data['num_vehicles'], data['depot'])

    # Create Routing Model.
    routing = pywrapcp.RoutingModel(manager)


    # Create and register a transit callback.
    def distance_callback(from_index, to_index):
        """Returns the distance between the two nodes."""
        # Convert from routing variable Index to distance matrix NodeIndex.
        from_node = manager.IndexToNode(from_index)
        to_node = manager.IndexToNode(to_index)
        return data['distance_matrix'][from_node][to_node]

    transit_callback_index = routing.RegisterTransitCallback(distance_callback)

    # Define cost of each arc.
    routing.SetArcCostEvaluatorOfAllVehicles(transit_callback_index)

    # Add Distance constraint.
    dimension_name = 'Distance'
    routing.AddDimension(
        transit_callback_index,
        0,  # no slack
        10000,  # vehicle maximum travel distance
        True,  # start cumul to zero
        dimension_name)
    distance_dimension = routing.GetDimensionOrDie(dimension_name)
    distance_dimension.SetGlobalSpanCostCoefficient(100)

    # Setting first solution heuristic.
    search_parameters = pywrapcp.DefaultRoutingSearchParameters()
    search_parameters.first_solution_strategy = (
        routing_enums_pb2.FirstSolutionStrategy.PATH_CHEAPEST_ARC)
    search_parameters.local_search_metaheuristic = (
        routing_enums_pb2.LocalSearchMetaheuristic.TABU_SEARCH)
    search_parameters.time_limit.seconds = 60 * 60 * 4 #4 hours
    search_parameters.log_search = True

    # Solve the problem.
    solution = routing.SolveWithParameters(search_parameters)

    # Print solution on console.
    pdb.set_trace()

    if solution:
        print_solution(data, manager, routing, solution)
    else:
        print("No solution")


if __name__ == '__main__':
    main()
```
**Notes:**
 - distance matrix is 72*72 so 72 nodes 
 - I have made a very large distance between some nodes because I want to forbid such trips.
 - I have tried to run it without metahuristics and it's taking too much time to find a first solution, while overheating
- With 12 nodes, finding an initial solution is almost instant

I'm really desesperated as solving this problem is very important. Thank you for your help.

**Comment by lperron** [comment]
Please try all initial solution heuristics.

Le mer. 22 mai 2019 à 03:26, Souames Annis <notifications@github.com> a
écrit :

**Comment by GuyBenhaim** [comment]
On a similar issue: does including 'infinite' value in certain Node_A, Node_B combination in the distance matrix cause the solver to skip testing any other contraints, and run faster?
Of course, we could repeat this for many irrelevant tasks/arcs.

**Comment by anis-coding** [comment]
> Please try all initial solution heuristics. Le mer. 22 mai 2019 à 03:26, Souames Annis 

@lperron  Thank's for your advice. I tried several algorithms for first solution strategy, I found that CHRISTOFIDES is the best one in my case processing 170 nodes in 40 seconds. PARALLEL_CHEAPEST_INSERTION is good too.

Now I'm trying to improve the solution using TABU search, but I'm having a weird result: 

I left TABU search running for an hour, but after that, the sum of all routes distances is around 35 000 KM using 23 vehicles out of 70 vehicle. While without using TABU Search (Using CHRISTOFIDES only) I got 30 000 KM as total distance with 19 vehicles only, how is that possible ? 

However TABU search did return a smaller maximum route distance (1700 KM while without TABU search the maximum route distance (cycle) was 2200 KM). 

My goal is to use the minimum number of vehicles and reduce the overall distance across the whole trip (not just a cycle).

So what's going wrong ?
